public/components/visualizations/visualization.tsx

```diff
@@ -29,14 +29,14 @@ export const Visualization = ({
             [GROUPBY]: dimensions = [],
             [AGGREGATIONS]: series = [],
             queryMetaData = {},
-            subType = '',
+            metricType = '',
           } = {},
         } = {},
       },
       vis = {},
     }: IVisualizationContainerProps = vs;
 
-    if (subType === PROMQL_METRIC_SUBTYPE) {
+    if (metricType === PROMQL_METRIC_SUBTYPE) {
```

Comment:
I see a lot of changes from subType to metricType. Has backward compatibility been tested manually, for the prom-metrics that were created in 2.11 and saved in explorer.

Reply:
All the promql metrics created in 2.11 would have the subType as `metric`. There was no classification between prometheus metrics and custom metrics back then. This `promql type` in `subType` was introduced in Peter's PR merged couple of weeks back . I'm completely removing it and moving it to `metricType` so that `subType` isn't modified in anyways. This ensures backward compatibility as well. If we introduce `promql` metric type in subType, that would break the backward compatibility.

Additionally metricType is to store which kind of metric is being stored in backend so that it's easier to differentiate between metrics and also not modify the current subType to anything other default '`metric`'

Peter's PR for reference - https://github.com/opensearch-project/dashboards-observability/pull/1305/files#diff-94efc963006c9807804dc16917c8e0445d5a342cfc84e16cdeae9a5f185b4d59R82